src/components/hv-view/index.js

```diff
@@ -41,142 +47,213 @@ export default class HvView extends PureComponent<HvComponentProps> {
+    const defaultOffset = 120;
+    if (this.attributes[ATTRIBUTES.SCROLL_TO_INPUT_OFFSET]) {
+      const offset = parseInt(
+        this.attributes[ATTRIBUTES.SCROLL_TO_INPUT_OFFSET],
+        10,
+      );
+      return Number.isNaN(offset) ? 0 : defaultOffset;
     }
+    return defaultOffset;
+  };
+
+  getKeyboardAwareScrollViewProps = (
+    inputFieldRefs: Array<any>,
+  ): KeyboardAwareScrollViewProps => ({
+    automaticallyAdjustContentInsets: false,
+    getTextInputRefs: () => inputFieldRefs,
+    keyboardShouldPersistTaps: 'handled',
+    scrollEventThrottle: 16,
+    scrollToInputAdditionalOffset: this.getScrollToInputAdditionalOffsetProp(),
+  });
+
+  Content = () => {
+    /**
+     * Useful when you want keyboard avoiding behavior in non-scrollable views.
+     * Note: Android has built-in support for avoiding keyboard.
+     */
+    const keyboardAvoiding =
+      this.attributes[ATTRIBUTES.AVOID_KEYBOARD] === 'true' &&
+      Platform.OS === 'ios';
 
+    const hasInputFields = this.hasInputFields();
+    const inputFieldRefs = [];
+    const scrollable = this.attributes[ATTRIBUTES.SCROLL] === 'true';
+    const safeArea = this.attributes[ATTRIBUTES.SAFE_AREA] === 'true';
     if (safeArea) {
-      if (safeAreaIncompatible) {
+      if (keyboardAvoiding || scrollable) {
         console.warn('safe-area is incompatible with scroll or avoid-keyboard');
-      } else {
-        c = SafeAreaView;
       }
     }
 
     const children = Render.renderChildren(
       this.props.element,
       this.props.stylesheets,
       this.props.onUpdate,
-      viewOptions,
+      {
+        ...this.props.options,
+        ...(scrollable && hasInputFields
+          ? {
+              registerInputHandler: ref => {
+                if (ref !== null) {
+                  inputFieldRefs.push(ref);
+                }
+              },
+            }
+          : {}),
+      },
     );
 
-    if (scrollable && !horizontal) {
-      // add sticky indicies
-      const stickyIndices = children.reduce(
-        (acc, element, index) =>
-          typeof element !== 'string' &&
-          element.props?.element?.getAttribute('sticky') === 'true'
-            ? [...acc, index]
-            : acc,
-        [],
-      );
-      if (stickyIndices.length) {
-        props.stickyHeaderIndices = stickyIndices;
+    /* eslint-disable react/jsx-props-no-spreading */
+    if (scrollable) {
+      if (hasInputFields) {
+        return (
+          <KeyboardAwareScrollView
+            {...{
+              ...this.getCommonProps(),
+              ...this.getScrollViewProps(children),
+              ...this.getKeyboardAwareScrollViewProps(inputFieldRefs),
+            }}
+          >
+            {children}
+          </KeyboardAwareScrollView>
+        );
       }
+      return (
+        <ScrollView
+          {...{
+            ...this.getCommonProps(),
+            ...this.getScrollViewProps(children),
+          }}
```

Comment:
this spread usage looks a little funny, would the following return the same structure?
```suggestion
          {
            ...this.getCommonProps(),
            ...this.getScrollViewProps(children),
          }
```

Reply:
the surrounding curly brackets here are not representing a dictionary, but rather a [placeholder in a JSX template](https://reactjs.org/docs/introducing-jsx.html#embedding-expressions-in-jsx). Here, we spread a dictionary that's created on the fly with two other dictionaries (returned by the two function calls)